snapcraft/cli/store.py

```diff
@@ -101,12 +106,21 @@ def push(snap_file, release):
     If --release is used, the channel map will be displayed after the
     operation takes place.
 
+    If --only-metadata is given, only the snap's metadata information will be
+    pushed to the server (not the whole .snap file). Because of this, --release
+    is not a compatible option to use with it.
+
     \b
     Examples:
         snapcraft push my-snap_0.1_amd64.snap
+        snapcraft push my-snap_0.1_amd64.snap --only-metadata
         snapcraft push my-snap_0.2_amd64.snap --release edge
         snapcraft push my-snap_0.3_amd64.snap --release candidate,beta
     """
+    if release and only_metadata:
+        raise click.UsageError(
+            "--only-metadata and --release are mutually exclusive")
```

Comment:
Can't this be done with click itself?

Reply:
I searched for this, it look it [can't be done and no plans for click to gain this capability](https://github.com/pallets/click/issues/257).